var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -56,23 +57,34 @@ class Gcc(AutotoolsPackage):
     version('4.6.4', 'b407a3d1480c11667f293bfb1f17d1a4')
     version('4.5.4', '27e459c2566b8209ab064570e1b378f7')
 
+    # Builds all default languages by default.
+    # Ada, Go, Jit, and Objective-C++ are not default languages.
+    # In that respect, the name 'all' is rather misleading.
+    variant('languages',
+            default='all',
```

Comment:
> Oh, wait. So before GCC 4.8, c wasn't a valid language. I'm sure it was still built, but languages=c will fail.

maybe one just filters it out from the configure options in such case? It is still installed, obviously. So from Spack's perspective `c` is a valid options for `@:4.8`. Otherwise you would have someone who installs `gcc@4.7 language=c` and it won't build.